frontend/src/components/DefenceBox/DefenceBox.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import { useEffect, useState } from "react";
+import "./DefenceBox.css";
+import DefenceMechanism from "./DefenceMechanism";
+import {
+  getActiveDefences,
+  activateDefence,
+  deactivateDefence,
+  DEFENCE_DETAILS,
+} from "../../service/defenceService";
+
+function DefenceBox({ triggeredDefences }: { triggeredDefences: string[] }) {
+  // list of defence mechanisms
+  const [defenceDetails, setDefenceDetails] = useState(
+    DEFENCE_DETAILS.map((defence) => {
+      return {
+        ...defence,
+        isActive: false,
+        isTriggered: false,
+      };
+    })
```

Comment:
If we have `isActive` and `isTriggered` default to `false` in the interface definition, we don't need this `map` here.